### PR TITLE
Initial version

### PR DESCRIPTION
**Pull request by dwillmer**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+  Contents
+} from 'jupyter-js-services';
+
+import {
+  Widget
+} from 'phosphor-widget';
+
+import './index.css';
+
+/**
```
Comment (blink1073) [inline src/index.ts:16]: Two blank lines.

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+   */
+  set onClick(cb: (name: string, contents: any) => void) {
+    this._onClick = cb;
+  }
+
+  /**
+   * Handle the events on the file browser.
+   */
+  handleEvent(event: Event): void {
+    if (!this.node.contains((<any>event).target)) {
```
Comment (blink1073) [inline src/index.ts]: Use `as` for casting.

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+  /**
+   * Set the onClick handler for the file browser.
+   */
+  set onClick(cb: (name: string, contents: any) => void) {
+    this._onClick = cb;
+  }
+
+  /**
+   * Handle the events on the file browser.
```
Comment (blink1073) [inline src/index.ts]: Suggest using the docstring here: https://github.com/phosphorjs/phosphor-splitpanel/blob/master/src/index.ts#L269.

File: src/index.ts [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        var parts = this._currentDir.split('/');
+        var parts = parts.slice(0, parts.length-2);
+        if (parts.length === 0) {
+          this._currentDir = '';
+        } else {
+          this._currentDir = parts.join('/') + '/';
+        }
+        this.listDir();
+      } else {
+        var path = this._currentDir + (<HTMLElement>event.target).textContent;
```
Comment (blink1073) [inline src/index.ts]: `as` operator.

File: src/index.ts [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+        this._contents.get(path, {type:"file"}).then((msg: any) => {
```
Comment (blink1073) [inline src/index.ts]: Add spaces around `{ }`.

File: src/index.ts [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+        });
+      }
+    }
+  }
+
+  /**
+   * Set the file browser contents to the items in the
+   * current directory.
+   */
+  listDir() {
```
Comment (blink1073) [inline src/index.ts]: Missing a return type.

File: src/index.ts [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+        if ((<any>msg).content[i].length) {
+          this._addItem((<any>msg).content[i].name + '/', true);
+        } else {
+          this._addItem((<any>msg).content[i].name, false);
+        }
+      }
+    });
+  }
+
+  private _addItem(text: string, isDirectory: boolean) {
```
Comment (blink1073) [inline src/index.ts]: Missing a return type.

File: src/index.ts [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+   * Get the onClick handler for the file browser.
+   */
+  get onClick(): (name: string, contents: any) => void {
+    return this._onClick;
+  }
+
+  /**
+   * Set the onClick handler for the file browser.
+   */
+  set onClick(cb: (name: string, contents: any) => void) {
```
Comment (blink1073) [inline src/index.ts:57]: Missing return type.

File: src/index.ts [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+        '<div class="list_container"></div>' +
+      '</div>'
+    );
+    return node;
+  }
+
+  /**
+   * Construct a new file browser widget.
+   */
+  constructor(baseUrl: string, currentDir: string) {
```
Comment (blink1073) [inline src/index.ts]: We should add optional `contents: IContents` to the constructor to allow for other implementations of `Contents`.

File: src/index.ts (hunk 8 above)
Comment (dwillmer) [inline src/index.ts:57]: This is a setter, they don't have return types.

File: src/index.ts [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+   * not be called directly by user code.
+   */
+  handleEvent(event: Event): void {
+    if (!this.node.contains((event as any).target)) {
+      return;
+    }
+    if (event.type === 'mousedown') {
+      var el = event.target as HTMLElement;
+      var text = el.textContent;
+      if (text[text.length - 1] === "/") {
```
Comment (afshin) [inline src/index.ts]: Should these be single quotes?

File: src/index.ts [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+        var parts = parts.slice(0, parts.length-2);
+        if (parts.length === 0) {
+          this._currentDir = '';
+        } else {
+          this._currentDir = parts.join('/') + '/';
+        }
+        this.listDir();
+      } else {
+        var path = this._currentDir + (event.target as HTMLElement).textContent;
+        this._contents.get(path, "file", {}).then((msg: any) => {
```
Comment (afshin) [inline src/index.ts]: Should these be single quotes?

File: src/index.ts [hunk 12]
```diff
@@ -0,0 +1,150 @@
...
+   * Handle the DOM events for the file browser.
+   *
+   * @param event - The DOM event sent to the panel.
+   *
+   * #### Notes
+   * This method implements the DOM `EventListener` interface and is
+   * called in response to events on the panel's DOM node. It should
+   * not be called directly by user code.
+   */
+  handleEvent(event: Event): void {
```
Comment (afshin) [inline src/index.ts:71]: In other packages, `handleEvent` is usually a simple switch that deploys to private, named event handlers with names like `_evtMouseDown`. Even though you're only handling a single event type, maybe that's appropriate here?

File: src/index.ts [hunk 13]
```diff
@@ -0,0 +1,150 @@
...
+  listDir(): void {
+    this.node.firstChild.lastChild.textContent = '';
+    if (this._currentDir.lastIndexOf('/') !== -1) {
+      this._addItem('..', true);
+    }
+
+    var path = this._currentDir.slice(0, this._currentDir.length - 1);
+    this._contents.listContents(path).then((msg: any) => {
+      for (var i = 0; i < msg.content.length; i++) {
+        if ((<any>msg).content[i].length) {
```
Comment (blink1073) [inline src/index.ts]: Use the `as` operator.

File: lib/index.css [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+| Copyright (c) Jupyter Development Team.
+| Distributed under the terms of the Modified BSD License.
+|----------------------------------------------------------------------------*/
+.jp-FileBrowser-Item {
+  color: #2F2F2F;
+  display: inline-block;
+  font: 14px Helvetica, Arial, sans-serif;
+}
+
+.jp-FileBrowser-ListItem > div {
```
Comment (blink1073) [inline lib/index.css:11]: I think this should be `jp-FileBrowser-list-item`, etc.

Comment (blink1073) [inline lib/index.css:11]: That appears to be the convention in other Phosphor repos, lowercase unless it is the name of a `class`, and use hyphen separators.

File: README.md [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+Navigate to `docs/index.html`.
+
+
+Supported Runtimes
+------------------
+
+The runtime versions which are currently *known to work* are listed below.
+Earlier versions may also work, but come with no guarantees.
+
+- Node 0.12.7+
```
Comment (blink1073) [inline README.md:69]: Not going to be supported, I think.

File: src/index.ts (hunk 8 above)
Comment (blink1073) [inline src/index.ts:57]: touchè, salesman.

**Pull request merged**